typing/typecore.ml

```diff
@@ -1954,6 +1954,9 @@ struct
       | Texp_extension_constructor _ -> Static
       | Texp_apply ({exp_desc = Texp_ident (_, _, vd)}, _)
         when is_ref vd -> Static
+      | Texp_apply (_,args)
+        when List.exists (function (_,None) -> true | _ -> false) args ->
+          Static
```

Comment:
To someone reading the code without the full typedtree structure in mind, it is very unclear what this case is about (`is_ref` above is self-explanatory). Maybe you could avoid the `when` here, have a single (non-ref) `Texp_apply` case, and have on the test a comment explaining what the test is about? (Or you could make the test an auxiliary function with a self-explanatory name.)

Reply:
> Or you could make the test an auxiliary function with a self-explanatory name.

I picked this option, so that the named function could be reused in the two locations where it's needed.